packages/backend-deployer/src/cdk_error_mapper.ts

```diff
@@ -154,6 +213,16 @@ export class CdkErrorMapper {
       errorName: 'BackendBuildError',
       classification: 'ERROR',
     },
+    {
+      // We capture the parameter name to show relevant error message
+      errorRegex:
+        /Failed to retrieve backend secret (.*) for.*ParameterNotFound/,
+      humanReadableErrorMessage: `The secret ${this.placeHolder} specified in the backend does not exist.`,
```

Comment:
could be overkill at this point, but maybe we could use [named capturing groups](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Named_capturing_group) and replace all `{captureGroupName}` placeholders with the matching value?

If we did, this would become
```suggestion
        /Failed to retrieve backend secret (?<secretName>.*) for.*ParameterNotFound/,
      humanReadableErrorMessage: `The secret {secretName} specified in the backend does not exist.`,
```
IMO this is a little more readable to figure out what value is being mapped where.

Reply:
It's a good idea as this would only expand going forward. For now I'll cut a tech debt and take this over in a future PR.

Reply:
https://github.com/aws-amplify/amplify-backend/issues/1271